addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -493,7 +514,14 @@ export class Rtc {
                 }
                 break;
             case "trackChange": {
-                const { isSelfMuted, isTalking, isSendingVideo, isDeaf } = payload.state;
+                if (Object.values(this.mapTransceiverToMid).length === 0) {
+                    const transceivers = session.peerConnection.getTransceivers();
+                    for (let i = 0; i < transceivers.length; i++) {
+                        this.mapTransceiverToMid[ORDERED_TRANSCEIVER_NAMES[i]] =
+                            transceivers[i].mid;
```

Comment:
It looks like the mapping is not really useful because it assumes that you already know that indices of `ORDERED_TRANSCEIVER_NAMES` and of `transceivers` are matching.

So it may be possible to make less diff and do a function that looks like the inverse of `rtcService.getTransceiver()` (returns a label based on transceiver input by matching indices) that you would call in `onTrack` (or `updateStream`) to get the label based on the transceiver. And you would index all streams by label instead of `mid` in `session.videoStreams`